src/d/actor/d_a_obj_swheavy.cpp

```diff
@@ -4,133 +4,457 @@
 //
 
 #include "d/actor/d_a_obj_swheavy.h"
+#include "d/res/res_hhbot.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
+#include "d/actor/d_a_player.h"
+
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+const char daObjSwheavy::Act_c::M_arcname[] = "Hhbot";
+const daObjSwheavy::Act_c::Attr_c daObjSwheavy::Act_c::M_attr[4] = {
+    {
+        {},
+        0,
+        0.0f,
+        0.0f,
+        0.0f,
+        0,
+        0,
+        0,
+    },
+    {
+        {},
+        0,
+        0.0f,
+        0.0f,
+        0.0f,
+        0,
+        0,
+        0,
+    },
+    {
+        {0,0,0x80,0},
+        0xC,
+        0.93f,
+        0.42f,
+        0.0f,
+        4,
+        4,
+        15,
+    },
+    {
+        {0,0,0x80,0},
+        0xB,
+        0.93f,
+        0.42f,
+        0.0f,
+        4,
+        4,
+        15,
+    },
+};
 
 /* 00000078-0000009C       .text solidHeapCB__Q212daObjSwheavy5Act_cFP10fopAc_ac_c */
-void daObjSwheavy::Act_c::solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+BOOL daObjSwheavy::Act_c::solidHeapCB(fopAc_ac_c* i_this) {
+    return static_cast<daObjSwheavy::Act_c*>(i_this)->create_heap() & 0xFF; // Fakematch?
```

Comment:
I tried messing with return types to get this to match without the & 0xFF - found a similar thing in d_a_obj_akabe